tests/Unit/Controllers/SettingsControllerTest.php

```diff
@@ -10,34 +10,237 @@
 use PHPUnit\Framework\MockObject\MockObject;
 use Symfony\Component\HttpFoundation\Session\Session;
 use Symfony\Component\HttpFoundation\Session\Storage\MockArraySessionStorage;
+use Psr\Log\LoggerInterface;
+use Engelsystem\Helpers\Authenticator;
+use Engelsystem\Test\Unit\HasDatabase;
+use Engelsystem\Http\Request;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Log\Test\TestLogger;
+use Engelsystem\Http\UrlGeneratorInterface;
+use Engelsystem\Http\UrlGenerator;
+use Engelsystem\Models\User\User;
 
 class SettingsControllerTest extends TestCase
 {
+    use HasDatabase;
+
+    /** @var Authenticator|MockObject */
+    protected $auth;
+
+    /** @var Config */
+    protected $config;
+
+    /** @var TestLogger */
+    protected $log;
+
+    /** @var Response|MockObject */
+    protected $response;
+
+    /** @var Request */
+    protected $request;
+
+    /** @var User */
+    protected $user;
+
+    /**
+     * @covers \Engelsystem\Controllers\SettingsController::password
+     */
+    public function testPassword()
+    {
+        /** @var Response|MockObject $response */
+        $this->response->expects($this->once())
+        ->method('withView')
+        ->willReturnCallback(function ($view, $data) {
+            $this->assertEquals('pages/settings/password.twig', $view);
+
+            return $this->response;
+        });
+
+        /** @var SettingsController $controller */
+        $controller = $this->app->make(SettingsController::class);
+        $controller->password();
+    }
+
+    /**
+     * @covers \Engelsystem\Controllers\SettingsController::savePassword
+     */
+    public function testSavePassword()
+    {
+        $body = [
+            'password' => 'password',
+            'new_password' => 'newpassword',
+            'new_password2' => 'newpassword'
+        ];
+        $this->request = $this->request->withParsedBody($body);
+
+        $this->auth->expects($this->once())
```

Comment:
:information_source: There is a `$this->setExpects(...)` method which shrinks it to one line